test/tests/geomsearch/quadrature_locator_plus_constraint/tests

```diff
@@ -0,0 +1,9 @@
+[Tests]
+  [quadrature_locator_plus_constraint]
+    type = RunApp
+    input = quadrature_locator_plus_constraint.i
+    issues = '#5234'
+    requirement = 'We shall be able to augment our sparsity based on constraints even when we have a quadrature nearest-node locator. The key is to not to try to access dof information when generating the purely geometric quadrature nodes (we may not have even initialized the DofMap at this point)'
```

Comment:
What the heck is going on with this requirement!? On one hand the detail for the programmer is great. However, it's not the best requirement following these definitions: https://www.mooseframework.org/sqa/what_is_a_requirement.html

Reply:
Since all I'm doing right now is cleanup up bad requirements. I'd like people to get better at not creating those in the first place. Use The system, not "we". Also try to avoid compound statements. This is like a mini-paragraph. I think the first sentence may be reasonable for the requirement here except I'd probably drop the word even. It's just a statement of what the system should do. The rest of this stuff could be put into the md, or even just a comment in the test spec or input file.

Reply:
>  https://www.mooseframework.org/sqa/what_is_a_requirement.html

The big blue note says that the requirement should be unambiguous. I would say that this is very unambiguous!

Reply:
But I'll change it